addons/binding/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/handler/PHCBridgeHandler.java

```diff
@@ -0,0 +1,693 @@
+                return;
+            }
+
+            if (buffer.length - 1 - pos < 3) {
+                messageFragment = new byte[buffer.length - pos + 1];
+                System.arraycopy(buffer, messageStart, messageFragment, 0, messageFragment.length);
+            }
+
+            byte size = 0;
+            result.add(buffer[pos]);
+            size = (byte) (buffer[pos] & 0x7F);
+
+            if (size > 0 && size < 4) {
+                if (buffer.length - 1 - pos < size + 2) {
+                    messageFragment = new byte[buffer.length - pos + 1];
+                    System.arraycopy(buffer, messageStart, messageFragment, 0, messageFragment.length);
+                }
+
+                byte[] message = new byte[size + 4];
+                message[0] = moduleAddress;
+                message[1] = buffer[pos++];
+
+                for (int i = 2; i < size + 4; i++) {
+                    message[i] = buffer[pos++];
+                    result.add(message[i]);
+                }
+
+                if (!repeat.contains(Arrays.toString(message))) { // avoids multiple processing of the same message
+                    processIncomingMessage(message, size);
+                    repeat.add(Arrays.toString(message));
+                }
+            }
+        }
+    }
+
+    private void processQueueLoop() {
+        while (true) {
+            long limit = System.currentTimeMillis();
+
+            Map<Long, QueueObject> subQueue = queue.subMap(0L, true, limit, true);
+
+            for (Long key : subQueue.keySet()) {
+                QueueObject qo = subQueue.get(key);
+                queue.remove(key);
+                if (qo.getCounter() < SEND_RETRY_COUNT && !qo.getCommand()
+                        .equals(isChannelOutputStateSet(qo.getModuleType(), qo.getModuleAddress(), qo.getChannel()))) {
+                    qo.increaseCounter();
+
+                    switch (qo.getModuleType()) {
+                        case PHCBindingConstants.CHANNELS_AM:
+                            sendAm(qo.getModuleAddress(), qo.getChannel(), qo.getCommand());
+                            break;
+                        case PHCBindingConstants.CHANNELS_EM_LED:
+                            sendEm(qo.getModuleAddress(), qo.getChannel(), qo.getCommand());
+                            break;
+                        case PHCBindingConstants.CHANNELS_JRM:
+                            sendJRM(qo.getModuleAddress(), qo.getChannel(), qo.getCommand(), qo.getUpDownTime());
+                            break;
+                    }
+
+                    queue.put(System.currentTimeMillis() + SEND_RETRY_TIME, qo);
+
+                } else if (qo.getCounter() >= SEND_RETRY_COUNT
+                        && !qo.getModuleType().equals(PHCBindingConstants.CHANNELS_JRM)) {
+                    // Can´t process the acknowledgement of JRM yet.
+                    // TODO: If no acknowledge from the module received, perhaps set the status of the appropriate thing
```

Comment:
I would remove it because the communication isn't accurate enough, so I can't definitly say if the module is offline or not. Maybe can integrate it later, in the next step(PR) I try to get a more accurate serial communication.